dashboard/app/views/home/update_login.html.haml

```diff
@@ -0,0 +1,8 @@
+- if current_user
+  :javascript
+    dashboard.clientState.setCurrentUserKey('#{current_user.id}');
+- else
+  :javascript
+    dashboard.clientState.setAnonymousUser();
+:javascript
+  document.location = '#{@redirect}'
```

Comment:
For down the road, wonder if we might need a meta tag to harden this similar to #5847 (though in that case it was a bit of a "kill it with fire" super-edge-case without a repro)